### PR TITLE
docs: add docs for `DataFrame` and `Series` dunder methods

### PR DESCRIPTION
**Pull request by shobsi**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
  - `__add__`: https://screenshot.googleplex.com/8ZLiQJLZPTWEZdg
  - `__radd__`: https://screenshot.googleplex.com/5Rn7Bn8guqBrBTD
  - `radd`: https://screenshot.googleplex.com/C5KVoZCNTL9oiHq

Fixes internal issue 304747371 🦕

### CODE REVIEW

File: bigframes/dataframe.py [hunk 1]
```diff
@@ -797,11 +798,12 @@ def _apply_dataframe_binop(
     def eq(self, other: typing.Any, axis: str | int = "columns") -> DataFrame:
         return self._apply_binop(other, ops.eq_op, axis=axis)
 
+    def __eq__(self, other) -> DataFrame:
+        return self.eq(other)
+
```
Comment (ashleyxuu) [inline bigframes/dataframe.py:803]: how about adding `inspect.getdoc(vendored_pandas_frame.DataFrame.__eq__)` to generate the docs?

Comment (ashleyxuu) [reply bigframes/dataframe.py:803]: __eq__ = eq

**Review by tswast — COMMENTED** [review]
Were any of these docstring or examples derived from the pandas documentation? If so, they need to be moved to third_party.

**Comment by shobsi** [comment]
> Were any of these docstring or examples derived from the pandas documentation? If so, they need to be moved to third_party.

`DataFrame.__array__` is one case that could be debated. There is no pandas `DataFrame.__array__` documentation, so I put BigFrames `DataFrame.__array__` directly. There is pandas `Series.__array__`, so I added BigFrames `Series.__array__` in third party. For consistency I kept wording of `DataFrame.__array__` close to `Series.__array__`. Is that acceptable?

File: bigframes/dataframe.py (hunk 1 above)
Comment (shobsi) [reply bigframes/dataframe.py:847]: use this pattern at bunch of places, however for `__eq__` I created its own docstring with code samples.

**Comment by tswast** [comment]
> adding docstring with code samples directly in our classes which are worded similar to their public versions 

Anything derived from third-party content needs to be in the third-party directory.

**Review by ashleyxuu — APPROVED** [review]

File: bigframes/dataframe.py (hunk 1 above)
Comment (shobsi) [reply bigframes/dataframe.py:844]: Moved all dunder docstrings to third_party for compliance reasons (even slight similarity with an existing pandas documentation may be considered as derived) and used this pattern for explicit override.

**Comment by shobsi** [comment]
> > adding docstring with code samples directly in our classes which are worded similar to their public versions
> 
> Anything derived from third-party content needs to be in the third-party directory.

Done.

**Pull request merged**